### ISSUE TITLE
CreateLayer fails if SpatialReference arg exists but is invalid (undef or blank)

### ISSUE THREAD

**Issue by shawnlaffan** [ajolma/Geo-GDAL-FFI#34]
Geo::GDAL::FFI::DataSet::CreateLayer calls OSRClone on the SpatialReference arg irrespective of its value.  This then crashes when the value is undef or ''.

https://github.com/ajolma/Geo-GDAL-FFI/blob/66747347ca320a1cd855f373e686eba9c2ecfc7a/lib/Geo/GDAL/FFI/Dataset.pm#L115

The check could instead test for the boolean value of the arg.  

```perl
        $sr = Geo::GDAL::FFI::OSRClone(${$args->{SpatialReference}}) if $args->{SpatialReference};
```

**Comment by ajolma** [comment]
Turns out I have not understood autovivification correctly. "if" does not autovivify.

**Comment by shawnlaffan** [comment]
Autoviv generates intermediate data structures, but does not generate top level entries.  So, ```$hash->{sub1}{sub2}``` is like ```$hash->{sub1} = {} and $hash->{sub1}{sub2}```, but ```$hash->{sub1}``` does not create an entry.

In any case, the clone method should not be called if ```$args->{SpatialReference}``` is undef or otherwise invalid.

**Comment by ajolma** [comment]
Fixed in f0245f9
